File: backend/app/schemas/extras.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from app.models.enums import (
    ContentType,
    NotificationStatus,
    NotificationType,
    RecommendationCategory,
)


class RecommendationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    category: RecommendationCategory
    title_ar: str
    content_ar: str
    media_url: Optional[str] = None


class PatientRecommendationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    recommendation: RecommendationOut
    delivered_at: datetime
    is_viewed: bool
    is_helpful_feedback: Optional[bool] = None


class RecommendationFeedback(BaseModel):
    is_helpful: bool


class RecommendationCreate(BaseModel):
    category: RecommendationCategory
    chronic_condition_id: Optional[uuid.UUID] = None
    applicable_risk_levels: list[int]
    title_ar: str
    content_ar: str
    media_url: Optional[str] = None


class AdminRecommendationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    category: RecommendationCategory
    chronic_condition_id: Optional[uuid.UUID] = None
    applicable_risk_levels: list[int]
    title_ar: str
    content_ar: str
    media_url: Optional[str] = None
    is_active: bool
    created_at: datetime


class RecommendationUpdate(BaseModel):
    category: Optional[RecommendationCategory] = None
    chronic_condition_id: Optional[uuid.UUID] = None
    applicable_risk_levels: Optional[list[int]] = None
    title_ar: Optional[str] = None
    content_ar: Optional[str] = None
    media_url: Optional[str] = None
    is_active: Optional[bool] = None


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    type: NotificationType
    title_ar: str
    body_ar: str
    is_read: bool
    status: NotificationStatus
    created_at: datetime


class DeviceRegisterRequest(BaseModel):
    fcm_token: str
    device_type: Optional[str] = None


class ContentLibraryCreate(BaseModel):
    content_type: ContentType
    key: str
    chronic_condition_id: Optional[uuid.UUID] = None
    title_ar: Optional[str] = None
    body_ar: str
    is_published: bool = True


class ContentLibraryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    content_type: ContentType
    key: str
    title_ar: Optional[str] = None
    body_ar: str
    is_published: bool

File: mobile/lib/data/models/extras.dart
class Recommendation {
  final String id;
  final String category;
  final String titleAr;
  final String contentAr;
  final String? mediaUrl;

  Recommendation({
    required this.id,
    required this.category,
    required this.titleAr,
    required this.contentAr,
    this.mediaUrl,
  });

  factory Recommendation.fromJson(Map<String, dynamic> json) {
    return Recommendation(
      id: json["id"] as String,
      category: json["category"] as String,
      titleAr: json["title_ar"] as String,
      contentAr: json["content_ar"] as String,
      mediaUrl: json["media_url"] as String?,
    );
  }
}

class PatientRecommendation {
  final String id;
  final Recommendation recommendation;
  final DateTime deliveredAt;
  final bool isViewed;
  final bool? isHelpfulFeedback;

  PatientRecommendation({
    required this.id,
    required this.recommendation,
    required this.deliveredAt,
    required this.isViewed,
    this.isHelpfulFeedback,
  });

  factory PatientRecommendation.fromJson(Map<String, dynamic> json) {
    return PatientRecommendation(
      id: json["id"] as String,
      recommendation: Recommendation.fromJson(json["recommendation"] as Map<String, dynamic>),
      deliveredAt: DateTime.parse(json["delivered_at"] as String),
      isViewed: json["is_viewed"] as bool,
      isHelpfulFeedback: json["is_helpful_feedback"] as bool?,
    );
  }
}

class AppNotification {
  final String id;
  final String type;
  final String titleAr;
  final String bodyAr;
  final bool isRead;
  final String status;
  final DateTime createdAt;

  AppNotification({
    required this.id,
    required this.type,
    required this.titleAr,
    required this.bodyAr,
    required this.isRead,
    required this.status,
    required this.createdAt,
  });

  factory AppNotification.fromJson(Map<String, dynamic> json) {
    return AppNotification(
      id: json["id"] as String,
      type: json["type"] as String,
      titleAr: json["title_ar"] as String,
      bodyAr: json["body_ar"] as String,
      isRead: json["is_read"] as bool,
      status: json["status"] as String,
      createdAt: DateTime.parse(json["created_at"] as String),
    );
  }
}

class ContentLibraryItem {
  final String id;
  final String contentType;
  final String key;
  final String? titleAr;
  final String bodyAr;
  final bool isPublished;

  ContentLibraryItem({
    required this.id,
    required this.contentType,
    required this.key,
    this.titleAr,
    required this.bodyAr,
    required this.isPublished,
  });

  factory ContentLibraryItem.fromJson(Map<String, dynamic> json) {
    return ContentLibraryItem(
      id: json["id"] as String,
      contentType: json["content_type"] as String,
      key: json["key"] as String,
      titleAr: json["title_ar"] as String?,
      bodyAr: json["body_ar"] as String,
      isPublished: json["is_published"] as bool,
    );
  }
}
